src/wolfkrow/core/engine/task_graph.py

```diff
@@ -141,83 +170,37 @@ def export_tasks(self, export_type="Json", temp_dir=None, deadline=False):
                 deadline=deadline,
             )
 
-            # This is no longer true. We need a special case to handle chunked jobs now.
-            # # If the task export returns a list, that means it was a task which 
-            # # expanded into multiple other tasks. We need to add these new tasks 
-            # # to the task_graph.
-            # if isinstance(exported, list):
-            #     for exported_task in exported:
-            #         self.add_task(exported_task[0])
-            #     # Search the task graph for tasks which dependend on the original 
-            #     # task, and update them to depend on the new tasks.
-            #     for task2 in self._tasks.values():
-            #         if task.name in task2.dependencies:
-            #             for exported_task in exported:
-            #                 self.add_dependency(task2, exported_task[0].name)
-            # else:
-            #     exported = [exported]
-
-            exported_task_names = [export[0].name for export in exported]
+            exported_task_names = [export.task.task_name for export in exported]
 
             if len(exported) > 1:
                 for exported_task in exported[1:]:
-                    self.add_task(exported_task[0])
+                    self.add_task(exported_task.task, prefix=self._prefix)
                     # Update the new tasks to depend on the original task.
-                    self.add_dependency(exported_task[0], task.name)
+                    self.add_dependency(exported_task.task, task.name)
 
-                # Search the task graph for tasks which dependend on the original 
+                # Search the task graph for tasks which depended on the original 
                 # task, and update them to depend on the new tasks.
                 for task2 in list(self._tasks.values()):
                     # Do not add a dependency to yourself.
-                    if task2.name == task.name:
+                    if task2.task_name == task.task_name:
```

Comment:
This change should be reverted.

If there are 2 of the same task, but with a different prefix, then it will end up adding a dependency to itself (Because dependencies are added based off the name, not the full_name.